### PR TITLE
Allow admin to promote supervisors

### PR DESCRIPTION
**Pull request by rpg600**

### CODE REVIEW

File: src/AppBundle/Repository/CommitteeMembershipRepository.php [hunk 1]
```diff
@@ -148,6 +148,27 @@ public function countHostMembers(string $committeeUuid): int
     }
 
     /**
+     * Returns the number of supervisor members for the given committee.
```
Comment (hhamon) [inline src/AppBundle/Repository/CommitteeMembershipRepository.php]: Obvious comment, can be removed.

File: src/AppBundle/Controller/Admin/AdminCommitteeController.php [hunk 2]
```diff
@@ -89,6 +88,8 @@ public function changeRoleAction(Committee $committee, $userId, $role): Response
             if ($membership) {
                 if ('host' === $role) {
                     $membership->setPrivilege(CommitteeMembership::COMMITTEE_HOST);
+                } elseif ('supervisor' === $role && $this->get('app.committee.manager')->countCommitteeSupervisors($committee) < 1) {
```
Comment (hhamon) [inline src/AppBundle/Controller/Admin/AdminCommitteeController.php]: This business logic should be encapsulated in a "manager" service.

File: src/AppBundle/Entity/CommitteeMembership.php [hunk 3]
```diff
@@ -212,13 +225,13 @@ public function isPromotableHost(): bool
         return $this->isFollower();
     }
 
-    public function promote(): void
+    public function promote(string $privilege = self::COMMITTEE_HOST): void
```
Comment (hhamon) [inline src/AppBundle/Entity/CommitteeMembership.php]: This should be reverted I think.

File: src/AppBundle/Entity/CommitteeMembership.php [hunk 4]
```diff
@@ -212,13 +225,13 @@ public function isPromotableHost(): bool
...
     {
         if (!$this->isPromotableHost()) {
             throw CommitteeMembershipException::createNotPromotableHostPrivilegeException($this->uuid);
         }
 
-        $this->privilege = self::COMMITTEE_HOST;
+        $this->privilege = $privilege;
```
Comment (hhamon) [inline src/AppBundle/Entity/CommitteeMembership.php]: This should be reverted I think.

File: src/AppBundle/Repository/CommitteeMembershipRepository.php [hunk 5]
```diff
@@ -147,6 +147,24 @@ public function countHostMembers(string $committeeUuid): int
         return (int) $query->getSingleScalarResult();
     }
 
+    public function countSupervisorMembers(string $committeeUuid): int
+    {
+        $committeeUuid = Uuid::fromString($committeeUuid);
+
+        $qb = $this->createQueryBuilder('cm');
```
Comment (mykiwi) [inline src/AppBundle/Repository/CommitteeMembershipRepository.php]: `$qb` is useless, cast too:
```
return $this->createQueryBuilder('cm')
    ->select('COUNT(cm.uuid)')
    ->where('cm.committeeUuid = :committee')
    ->andWhere('cm.privilege = :privilege')
    ->setParameter('committee', (string) $committeeUuid)
    ->setParameter('privilege', CommitteeMembership::COMMITTEE_SUPERVISOR)
    ->getQuery()
    ->getSingleScalarResult();
```

File: src/AppBundle/Entity/CommitteeMembership.php [hunk 6]
```diff
@@ -147,6 +153,13 @@ public static function createForAdherent(
         );
     }
 
+    public static function checkPrivilege(string $privilege): void
+    {
+        if (!in_array($privilege, self::PRIVILEGES, true)) {
+            throw new \InvalidArgumentException("Invalid privilege $privilege");
```
Comment (mykiwi) [inline src/AppBundle/Entity/CommitteeMembership.php]: sprintf?

**Pull request merged**